modules/uncertainty/R/ensemble.R

```diff
@@ -281,14 +278,12 @@ for (input_tag in names(settings$run$inputs)) {
 
   
   # Get the workflow id
-  if (!is.null(settings$workflow$id)) {
-    workflow.id <- settings$workflow$id
-  } else {
-    workflow.id <- -1
-  }
+  workflow.id <- settings$workflow$id %||% -1
```

Comment:
I've never seen this operator before, and it's not the sort of thing one can easily google, so could you add a comment explaining what this step is doing.

Reply:
This is the scalar null coalescing operator, provided by base R since 4.4.0 and by {rlang} before that -- see `?"%||%"` for details. Basically `x %||% y` is "x unless it's null, in which case y"

Reply:
@mdietze 👍🏻 to adding a quick comment like "# x %||% y is "x unless it's null, in which case y" when it first appears in a file could be helpful.

Agreed that it's not obvious, but it has been used in a number of [tidyverse](https://github.com/search?q=%22%25%7C%7C%25%22+language%3AR+org%3Atidyverse&type=code) (as well as ropensci) packages.

Reply:
I agree it seems like a handy function, but I think how it works if far from common knowledge, so it doesn't hurt to improve the documentation, both of what is occurring here and (more importantly) why (i.e., why do we set unspecified workflow IDs to -1 instead of leaving them NULL)

Reply:
> why do we set unspecified workflow IDs to -1

This is a good question. Looks like @dlebauer added this line in 2013 or earlier -- David, will you suggest the right explanatory comment and/or change to the default?

Reply:
I don't recall the rationale other than to represent a database id value as an integer that will not occur in the database. 

But I think for consistency with below it would make sense to use the hash
  
```suggestion
  # if workflow$id is null, give it a unique value
  workflow.id <- settings$workflow$id %||% rlang::hash(settings)
```

Reply:
Maybe? The issue I see is that using hash will set different workflow IDs for every site in the workflow. For ensemble ID I see a reasonable interpretation of that ("We're going to analyze all runs from this site as one ensemble, ignoring any from other sites"), but I'm not thinking of an equivalent justification for workflow ID -- I lean toward thinking it's better to leave it clearly undefined at this per-site level if nobody set it at the whole-workflow level. Especially since it looks like printing workflow.id into README.txt is the only thing DB-free runs of write.ensemble.configs do with it.